File: factorial.c
#include <stdio.h>
#include <stdlib.h>

unsigned long long factorial(long);

int main(int argc, char *argv[])
{

    unsigned long long facto;
    long num = strtol(argv[1], NULL, 10);

    facto = factorial(num);

    printf("Factorial of %s is: %llu\n", argv[1], facto);

    return 0;
}

unsigned long long factorial(long num)
{
    if (num == 0 || num == 1)
        return 1;
    return (num * factorial(num - 1));
}

File: factorial.py
import sys
from timeit import default_timer as timer


def factorial(num):
    if num == 1 or 0:
        return 1
    else:
        return num * factorial(num - 1)


num = int(sys.argv[1])
start = timer()
facto = factorial(num)
end = timer()
elapse_time = end - start

print('\n')
print(f"-------------------------------------")
print(f"Factorial of {num}: {facto}\n")
print(f"Elapse time: {elapse_time}")
print(f"-------------------------------------")
print('\n')
